pulp_rpm/app/models/repository.py

```diff
@@ -316,6 +324,21 @@ def finalize_new_version(self, new_version):
             except RepositoryVersion.DoesNotExist:
                 previous_version = None
 
+        # If requested, validate that the added rpms are signed appropriately
+        if self.allowed_pub_keys:
+            rejected_nevras = []
+            if added := new_version.added().filter(pulp_type=Package.get_pulp_type()):
+                for package in added:
+                    package = package.cast()  # Transform it from Content into Package
```

Comment:
Don't do this, go the other way around

`Package.objects.filter(pk__in=new_version.added())`